src/test-utils/smoke-test-utils.ts

```diff
@@ -101,5 +153,50 @@ export const isEachEventIngested = async (
         if (!ingestedEvents.has(eventId)) {
             throw new Error(`Event ${eventId} not ingested.`);
         }
+
+        // validate custom attributes
+        if (metadataAttributes) {
+            // check for valid custom attributes
+            if (
+                !metadataAttributes?.every((flattenedAttribute) =>
+                    ingestedEvents.get(eventId)?.includes(flattenedAttribute)
+                )
+            ) {
+                console.log(
+                    `Expected attributes: ${JSON.stringify(metadataAttributes)}`
+                );
+                console.log(
+                    `Retrieved metadata: ${JSON.stringify(
+                        ingestedEvents.get(eventId)
+                    )}`
+                );
+                throw new Error(`Did not find expected metadata attribute(s).`);
+            }
```

Comment:
thought: I wonder if we could use *expect* expressions instead of  logging to console and throwing errors. Playwright's `expect` is simple but could work, or we could use a library like [expect](https://www.npmjs.com/package/expect).

Non-blocking. The code as-implemented is consistent with the existing code and works fine.